### PR TITLE
py-lazyarray: add new version 0.3.2

### PR DESCRIPTION
**Pull request by healther**
Part of our recent merge and upstream effort.

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
numpy 1.8+ is now required

**Comment by healther** [comment]
And also 1.12: if py3 and py26 support was dropped.

File: var/spack/repos/builtin/packages/py-lazyarray/package.py [hunk 1]
```diff
@@ -13,8 +13,12 @@ class PyLazyarray(PythonPackage):
     homepage = "http://bitbucket.org/apdavison/lazyarray/"
     pypi = "lazyarray/lazyarray-0.2.8.tar.gz"
 
+    version('0.3.2',  sha256='be980534c5950a976709085570f69be9534bdf0f3e5c21a9113de3ee2052683e')
     version('0.2.10', sha256='7a53f81b5f3a098c04003d2ad179fc197451fd96bc921510f8534c6af8cc8e19')
     version('0.2.8',  sha256='aaee4e18117cc512de7a4e64522f37bc6f4bf125ecffdbdbf4e4e390fbdd9ba2')
 
+    depends_on('python@2.7:', when='@0.3:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lazyarray/package.py]: Where do you see this requirement?

File: var/spack/repos/builtin/packages/py-lazyarray/package.py [hunk 2]
```diff
@@ -13,8 +13,12 @@ class PyLazyarray(PythonPackage):
...
     depends_on('py-numpy@1.3:', type=('build', 'run'))
+    depends_on('py-numpy@1.8:', type=('build', 'run'), when='@0.3:')
     depends_on('py-numpy@1.5:', type=('build', 'run'), when='^python@3:')
+    depends_on('py-numpy@1.12:', type=('build', 'run'), when='@0.3:^python@3:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lazyarray/package.py:26]: Where do you see this requirement?

File: var/spack/repos/builtin/packages/py-lazyarray/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-lazyarray/package.py]: It's in the changelog for 0.3.0 doc/installation.txt in [bb2f6f](https://github.com/NeuralEnsemble/lazyarray/commit/bb2f6fa3d9769c7b21fd357f46e60712b20e1c2b). Yes they do not tell that to setup.py but I assume that it is there for a reason.

File: var/spack/repos/builtin/packages/py-lazyarray/package.py (hunk 2 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-lazyarray/package.py:26]: see above (and yes they are not the best at documenting stuff in machine readable form)

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-lazyarray/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lazyarray/package.py]:
```suggestion
    # Required versions come from doc/installation.txt or:
    # https://lazyarray.readthedocs.io/en/latest/installation.html#dependencies
    depends_on('python@2.7:', when='@0.3:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**